crates/api/src/local_user/save_settings.rs

```diff
@@ -117,6 +119,21 @@ impl Perform for SaveUserSettings {
     })
     .await?
     .map_err(|e| LemmyError::from_error_message(e, "user_already_exists"))?;
+    let discussion_languages: Option<Vec<LanguageId>> =
+      if let Some(discussion_languages) = data.discussion_languages.clone() {
+        let mut language_ids = vec![];
+        for l in discussion_languages {
+          language_ids.push(
+            blocking(context.pool(), move |conn| {
```

Comment:
Reading this from the DB is no longer necessary, but you'll need to do upserts to the local_user_language table based on the provided language_ids.